### PR TITLE
OAK-11199 getSubject is supported only if a security manager is allowed

### PR DESCRIPTION
**Pull request by mbaedke**

### CODE REVIEW

**Review by reschke — CHANGES_REQUESTED** [review]
a) Compat*i*bility
b) The new package in commons would need a package-info

Otherwise, looks good.

**Comment by reschke** [comment]
Recommendation:

- new package commons.jdkcompat (non-public, "internal" in package-info), export in pom.xml
- name of util class more specific, like Java23Security

**Comment by mbaedke** [comment]
Don't know about the name Java23Security. It's certainly not more specific :)

**Review by reschke — APPROVED** [review]
The changes in commons now look good.

The other parts of the PR are minimal, which is good.

I haven't tested, I assume you did, @mbaedke 

The one thing that came to mind is that if we change Java23Security to "Subject", changes would be even less intrusive. Or even "Java23Subject"?  (this is bikeshedding, ignore if you want to)

**Review by reschke — COMMENTED** [review]
And maybe we should deal with

https://sonarcloud.io/component_measures?id=org.apache.jackrabbit%3Ajackrabbit-oak&pullRequest=1891&metric=new_coverage&view=list

**Comment by mbaedke** [comment]
I like Java23Subject. I wouldn't call it Subject at least for the reason that the method signatures are slightly different, which might confuse readers. Of course I tested, but will run integration tests again and look into the coverage issue before merging.

File: oak-security-spi/pom.xml [hunk 1]
```diff
@@ -109,6 +109,11 @@
       <artifactId>oak-jackrabbit-api</artifactId>
       <version>${project.version}</version>
     </dependency>
+    <dependency>
```
Comment (reschke) [inline oak-security-spi/pom.xml:112]: Duplicated dependency.

**Pull request merged**